packages/contracts/ts/deploy.ts

```diff
@@ -162,17 +169,33 @@ export const deployVerifier = async (signer?: Signer, quiet = false): Promise<Ve
 
 /**
  * Deploy a constant initial voice credit proxy contract
+ * @param args - the deploy constant initial voice credit proxy arguments
  * @param signer - the signer to use to deploy the contract
- * @param amount - the amount of initial voice credit to give to each user
  * @param quiet - whether to suppress console output
  * @returns the deployed ConstantInitialVoiceCreditProxy contract
  */
 export const deployConstantInitialVoiceCreditProxy = async (
-  amount: number,
+  args: { amount: number },
   signer?: Signer,
   quiet = false,
-): Promise<ConstantInitialVoiceCreditProxy> =>
-  deployContract<ConstantInitialVoiceCreditProxy>("ConstantInitialVoiceCreditProxy", signer, quiet, amount.toString());
+): Promise<[ConstantInitialVoiceCreditProxy, ConstantInitialVoiceCreditProxyFactoryContract]> => {
+  if (!signer) {
+    throw new Error("Signer is not provided");
+  }
+
+  const initialVoiceCreditProxyFactory = await deployContract<
```

Comment:
should we have a separate function that does the full deployment, factory and then contract? I'd expect the factory to be deployed once and the initialVoiceCreditProxy deployed many times with different args - so one function that given the factory address deploys the clone?